tests/integration/tests/kiali_urls_test.go

```diff
@@ -16,3 +19,93 @@ func TestKialiStatus(t *testing.T) {
 	assert.True(response)
 	assert.Equal(200, statusCode)
 }
+
+func TestKialiConfig(t *testing.T) {
+	assert := assert.New(t)
+	response, statusCode, err := utils.KialiConfig()
+
+	assert.Nil(err)
+	assert.Equal(200, statusCode)
+	assert.NotEmpty(response)
+	assert.NotEmpty(response.ClusterInfo)
+	assert.NotEmpty(response.ClusterInfo.Name)
+}
+
+func TestIstioPermissions(t *testing.T) {
+	assert := assert.New(t)
+	response, statusCode, err := utils.IstioPermissions()
+
+	assert.Nil(err)
+	assert.Equal(200, statusCode)
+	assert.NotNil(response)
+}
+
+func TestJaeger(t *testing.T) {
+	if os.Getenv("TEST_JAEGER") != "true" {
+		return
+	}
+	assert := assert.New(t)
+	response, statusCode, err := utils.Jaeger()
+
+	if statusCode == 200 {
+		assert.Nil(err)
+		assert.NotNil(response)
+		assert.True(response.Enabled)
+		assert.True(response.Integration)
+		assert.NotEmpty(response.URL)
+	} else {
+		assert.Fail(fmt.Sprintf("Status code should be '200' but was: %d and error: %s", statusCode, err.Error()))
+	}
+}
+
+func TestGrafana(t *testing.T) {
```

Comment:
It looks like grafana is deployed in CI but [this issue](https://github.com/kiali/kiali/issues/5030) might be why the api endpoint is failing. In the meantime, you'll need to explicitly setup grafana on the Kiali CR in the CI setup script. Setting the `url` field and/or `in_cluster_url` field should get it working.